packages/dd-trace/src/tracer.js

```diff
@@ -24,6 +25,12 @@ class DatadogTracer extends Tracer {
     this._scope = new Scope()
     setStartupLogConfig(config)
     flushStartupLogs(log)
+
+    try {
+      this._inmem_cfg = storeConfig(config)
+    } catch (error) {
+      log.warn(`Warning: Could not store tracer configuration for service discovery (Reason: ${error})`)
+    }
```

Comment:
As far as I can see, `new processDiscovery.TracerMetadata()` is not meant to throw if you provide it the required parameters, which you do. In those cases we normally don't surround statements with a try-catch block in JavaScript. So normally I'd write it something like this:

```suggestion
    this._inmem_cfg = storeConfig(config)
    if (this._inmem_cfg === undefined) {
      log.warn('Could not store tracer configuration for service discovery (Can\'t load process-discovery library)')
    }
```

And then update `storeConfig` to just return early in case `libdatadog.maybeLoad('process-discovery')` returns `undefined`